### PR TITLE
JIT: start checking sum of successor likelihoods

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add the ability to check that successor edge likelihoods sum to 1.0 Verify that successor edge likelihoods sum to 1.0 for the phases up through and including `fgAddInternal`.

Fixed a few problems
* importer branch folding needs to update likelihoods
* gdv likelihood needed some adjustments, especially for the multi-guess case
* profile incorporation for OSR methods also needed some fixing. Here we may have inconsistencies so the right answer is not always clear. With this change we now trust the successor edge counts even if they do not sum to the block count.

Contributes to #93020.

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=583204&view=ms.vss-build-web.run-extensions-tab)

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@amanasifkhalid PTAL
cc @dotnet/jit-contrib 

A few small diffs expected.

File: src/coreclr/jit/indirectcalltransformer.cpp
Comment (amanasifkhalid) [inline src/coreclr/jit/indirectcalltransformer.cpp:1032]: Is there a typo here? I think this evaluates to `0.1 / 0.3`, or `1/3`.

File: src/coreclr/jit/indirectcalltransformer.cpp
Comment (amanasifkhalid) [inline src/coreclr/jit/indirectcalltransformer.cpp:1045]: Since `baseLikelihood` is unsigned, are we just asserting that it's not zero here? If we're worried about underflow, then maybe we can flip the logic here by summing the likelihoods, and then checking that it's less than 100.

**Review by amanasifkhalid — COMMENTED** [review]
LGTM, I'll take another look once you've merged from main. Thanks!

File: src/coreclr/jit/fginline.cpp
Comment (AndyAyersMS) [reply src/coreclr/jit/fginline.cpp:688]: Yes, gone now.

File: src/coreclr/jit/indirectcalltransformer.cpp
Comment (AndyAyersMS) [reply src/coreclr/jit/indirectcalltransformer.cpp]: Yeah, fixed...

File: src/coreclr/jit/indirectcalltransformer.cpp
Comment (AndyAyersMS) [reply src/coreclr/jit/indirectcalltransformer.cpp:1046]: The values should sum to less than 100, so let me change this bit.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

**Comment by AndyAyersMS** [comment]
One runtime failure is #99074, another is (perhaps) #98817
Replays are hitting some sort of infra issue trying to download results from helix?
```
  Waiting for completion of job 0a0cdc60-4209-4427-8d83-4668d827952c on Windows.10.Amd64.Open (Details: https://helix.dot.net/api/jobs/0a0cdc60-4209-4427-8d83-4668d827952c/details?api-version=2019-06-17 )
  Job 0a0cdc60-4209-4427-8d83-4668d827952c on Windows.10.Amd64.Open is completed with 7 finished work items.
  Downloading result files for job 0a0cdc60-4209-4427-8d83-4668d827952c
D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\9.0.0-beta.24112.1\tools\download-results\DownloadFromResultsContainer.targets(26,5): 
   error : RestApiException`1: The response contained an invalid status code 500 Internal Server Error
   [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]

``

**Pull request merged**
